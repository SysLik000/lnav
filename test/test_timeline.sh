#! /bin/bash

export YES_COLOR=1

export TZ=UTC

touch -t 200711030923 ${srcdir}/logfile_glog.0
run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    -c ":hide-fields log_opid" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = 'test1' where log_line in (1, 3, 6)" \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = '1234' where log_line in (1, 3, 6)" \
    -c ";UPDATE all_opids SET description = 'test-1' WHERE opid = '1234'" \
    -c ";UPDATE all_logs set log_opid = '5678' where log_line = 2" \
    -c ";UPDATE all_opids SET description = 'test-2' WHERE opid = '5678'" \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ";UPDATE all_logs set log_opid = '1234' where log_line in (1, 3, 6)" \
    -c ";UPDATE all_opids SET description = 'test-1' WHERE opid = '1234'" \
    -c ";UPDATE all_logs set log_opid = '5678' where log_line = 2" \
    -c ";UPDATE all_opids SET description = 'test-2' WHERE opid = '5678'" \
    -c ";SELECT * FROM all_opids" \
    ${test_dir}/logfile_glog.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_generic.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':filter-in CdysLK1XpcrXOpVDuh' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':filter-out CdysLK1XpcrXOpVDuh' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-file *' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':close *' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-lines-before 2011-11-03 00:19:37' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    -c ':hide-lines-after 2011-11-03 00:20:30' \
    ${test_dir}/logfile_bro_http.log.0

run_cap_test ${lnav_test} -n \
    -c ':switch-to-view timeline' \
    ${test_dir}/logfile_strace_log.1
