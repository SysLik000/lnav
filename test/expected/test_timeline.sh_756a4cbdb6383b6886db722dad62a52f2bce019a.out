[1m[4m[35m   Duration   [0m[4m|[0m[4m [0m[1m[4m[31mâœ˜[0m[4m[33mâ–²[0m[4m [0m[4m|[0m[1m[4m[35m Operation[0m[4m [0m[4m [0m[4m|[0m[4m  10m    [0m[4m|[0m[4m  19m    [0m[4m|[0m[4m  29m    [0m[4m|[0m[4m  38m    [0m[4m|[0m[4m  48m    [0m[4m|[0m[4m  57m    [0m[4m|[0m[4m1h07m    [0m[4m|[0m[4m1h16m    [0m[4m|[0m[4m1h26m    [0m[4m|[0m[4m1h35m    [0m[4m|[0m[4m1h45m    [0m[4m|[0m[4m1h54m    [0m[4m|[0m[4m2h04m    [0m[4m|[0m[4m2h13m    [0m[4m|[0m[4m2h23m    [0m[4m|[0m[4m2h32m    [0m[4m|[0m[4m2h42m    [0m[4m|[0m[4m2h51m[0m
[32m           017[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32m75866859226275b[0m[7m[32m0[0m[32m4b5ac03d429cf9e1[0m[32m [0m[32m10.10.10.117[0m[32m facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)[0m
[32m         724us[0m[32m  [0m[1m[31m [0m[33m [0m[32m  [0m[32md584464393debb0e[0m[7m[32m8[0m[32m944f914f6e530a0[0m[32m [0m[32m10.10.10.134[0m[32m Mozilla/5.0 (compatible; AhrefsBot/7.0; +http://ahrefs.com/robot/)[0m
[1m[32m           024[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mcd4b289c5e6ef712[0m[1m[7m[32m6[0m[1m[32mef36a1fa5c6c6d5[0m[1m[32m [0m[1m[32m10.10.10.132[0m[1m[32m Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; bingbot/2.0; +http://www.bing.com/bingbot.htm) Chrome/116.0.1938.76 Safari/537.36[0m
[1m[32m         275us[0m[1m[32m  [0m[1m[31m [0m[1m[33m [0m[1m[32m  [0m[1m[32mbbf34c1c4279d18fe4da4f5648f05397[0m[1m[32m 240e:1:1::1234 Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/117.0.5938.60 Safari/537.36[0m
